packages/www/lib/utils/can-send-email.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Function to check if a user can request reset password or verfication email
+ */
+
+export function canSendEmail(emailType: "resetPassword" | "verifyEmail"): {
```

Comment:
I don't love that the frontend is responsible for doing rate limiting for reset password. This should be a responsibility of the backend, not the frontend. A user can easily clear their local storage.

Reply:
As much as i agree with you, it might be a little more work on backend compared to frontend. 

From Amrit, 
> usually, emails should get queued and processed independently. when clicking the button -> queue under the <user_id>.<action_hash> unique id, so that if same action is done multiple times (with same inputs), it wont trigger multiple emails.

So i think for now, we can keep it on frontend, just to avoid situation like this[1] but in future we can implement it directly in the backend. 

[1]:

![image](https://github.com/livepeer/studio/assets/56798748/ad6b9996-7308-442d-b5bb-451f2af3666c)